src/Statement/Grouper/StatementGrouper.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+interface StatementGrouper {
+
+	/**
+	 * @param StatementList $statements
+	 *
+	 * @return StatementList[] An associative array, mapping statement group identifiers to
```

Comment:
> "Is there something these identifiers have in common" is a specific question.

To what does "common" refer to? What do you expect? A common prefix or something? A predefined set of identifiers? They are nothing but array keys, as I already explained. Either strings or integers. And since this is an interface it can **not** know where this comes from. How could it? Why should it? What else could I say?

We two are going in circles, again and again, and it starts driving me crazy. You asking the same obscure questions again and again and me explaining the same obvious facts again and again. I would love to understand why this happens. I would love to avoid this, preferably not by simply ignoring what you write.

> The question I'm asking is about intend

What do you mean with "intend"? This is an interface to group statements, a.k.a. splitting a StatementList into multiple StatementLists. That's all. How this is done depends on the implementation and can not be described in the interface. How would such a description look like?

There. I did it again. I repeated myself. I spend an other hour on nothing but trying to answer a comment I still do not understand.